test/e2e/serial/workload_placement_test.go

```diff
@@ -240,6 +250,83 @@ var _ = Describe("[serial][disruptive][scheduler] workload placement", func() {
 			Expect(schedOK).To(BeTrue(), "pod %s/%s not scheduled with expected scheduler %s", updatedPod.Namespace, updatedPod.Name, schedulerName)
 		})
 	})
+
+	When("cluster has two feasible nodes with taint but only one has the requested resources on a single NUMA zone", func() {
+		timeout := 5 * time.Minute
+		BeforeEach(func() {
+			numOfNodeToBePadded := len(nrts) - 1
+
+			rl := corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("3"),
+				corev1.ResourceMemory: resource.MustParse("8G"),
+			}
+			err := padder.Nodes(numOfNodeToBePadded).UntilAvailableIsResourceList(rl).Pad(timeout)
+			Expect(err).ToNot(HaveOccurred())
+
+			nodes, err := e2enodes.GetWorkerNodes(fxt.Client)
+			Expect(err).ToNot(HaveOccurred())
+
+			t, _, err := taints.ParseTaints([]string{testTaint()})
+			Expect(err).ToNot(HaveOccurred())
+
+			for i := range nodes {
+				node := &nodes[i]
+				updatedNode, updated, err := taints.AddOrUpdateTaint(node, &t[0])
+				Expect(err).ToNot(HaveOccurred())
+				if updated {
+					node = updatedNode
+					klog.Infof("adding taint: %q to node: %q", t[0].String(), node.Name)
+					err = fxt.Client.Update(context.TODO(), node)
+					Expect(err).ToNot(HaveOccurred())
+				}
+			}
+		})
+
+		AfterEach(func() {
+			nodes, err := e2enodes.GetWorkerNodes(fxt.Client)
+			Expect(err).ToNot(HaveOccurred())
+
+			t, _, err := taints.ParseTaints([]string{testTaint()})
+			Expect(err).ToNot(HaveOccurred())
+
+			for i := range nodes {
+				node := &nodes[i]
+				updatedNode, updated, err := taints.RemoveTaint(node, &t[0])
+				Expect(err).ToNot(HaveOccurred())
+				if updated {
+					node = updatedNode
+					klog.Infof("removing taint: %q from node: %q", t[0].String(), node.Name)
+					err = fxt.Client.Update(context.TODO(), node)
+					Expect(err).ToNot(HaveOccurred())
+				}
+			}
+		})
+
+		It("[test_id:47594] should make a pod with a toleration land on a node with enough resources on a specific NUMA zone", func() {
+			testPod := objects.NewTestPodPause(fxt.Namespace.Name, "testpod")
+			pSpec := &testPod.Spec
+			if pSpec.Tolerations == nil {
+				pSpec.Tolerations = testToleration()
+			} else {
+				pSpec.Tolerations = append(pSpec.Tolerations, testToleration()...)
+			}
+
+			testPod.Spec.SchedulerName = schedulerName
+			cnt := &testPod.Spec.Containers[0]
+			requiredRes := corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("4"),
+				corev1.ResourceMemory: resource.MustParse("100Mi"),
+			}
+			cnt.Resources.Requests = requiredRes
+			cnt.Resources.Limits = requiredRes
+
+			err := fxt.Client.Create(context.TODO(), testPod)
+			Expect(err).ToNot(HaveOccurred())
+
+			_, err = e2ewait.ForPodPhase(fxt.Client, testPod.Namespace, testPod.Name, corev1.PodRunning, timeout)
```

Comment:
>shouldn't we check the pod lands on the node we expect?

I had that in mind, but how do I know which is the targeted node?
In order for that to be fulfilled, we need to mention explicitly (by name or label) which nodes are going to be saturated, making sure to exclude at least one (depending on the requirement of the test) and mark it as the targeted node.

It's definitely possible, but then, we're throwing the responsibility of verifying if a node is applicable to be a targeted node or a saturated node, on the user. 
IOW, the user will have to verify that node X has enough resources in order to declare it as the targeted node, and explicitly mention all the other nodes (by name or label) to get saturated.

OTOH the current approach is just mentioning how many nodes we want to saturate: 
https://github.com/openshift-kni/numaresources-operator/pull/143/commits/7c0e60a8e88cf10844e1b1c0b59ec69da6d3283d#diff-68784af7379bbc72fe22f9f4df729d2f7c6c4b564a4709c5768adf1a2ce5f378R263 

A combination of both approaches will be as follow:
1. We'll just mention how many nodes we want to saturate - As we're doing today.
2. The padder package will make sure to find the applicable nodes and saturate them according to the user request -  As  we're doing todayץ
3. The padder package will return the names of the saturate nodes - API change in the padder package.
4. The user will deduce from the list, which node isn't saturated and mark it as the targeted node - Change in the test logic.

WDYT?

>we should also probably check the pod was scheduled with the NUMA-aware scheduler

Will do

Reply:
> > Can't the test do that on its own?
> 
> From my perspective test is the user. I want to save the user from dealing with all these checks.

Well I'd be happy if we can, but I believe we can be a little more demanding if our user is a machine (running a test) vs a human. So the test code can (and probably should, because there's hardly a simpler way) do a bit more homework here.

> Let me see what I can do. One thing I'm concerned about is, that atm skipping the test in case of an incompatible cluster (not having 2 multi-NUMA nodes) is a real issue which is not related to the test itself, but to our internal resources issues (which you're aware of)

Skipping the test is a perfectly valid flow if the cluster isn't up to the expectations. OTOH not running the testsuite at all -never- it's a completely different story. But we can (and we actually should) address both things in parallel to cover more ground in less time.